core/unit_test/TestGraph.hpp

```diff
@@ -203,11 +213,50 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph),
 
   graph.submit(execution_space_instances.at(1));
 
-  Kokkos::deep_copy(execution_space_instances.at(1), count_host, count);
-  Kokkos::deep_copy(execution_space_instances.at(1), bugs_host, bugs);
-  execution_space_instances.at(1).fence();
-  ASSERT_EQ(1, count_host());
-  ASSERT_EQ(0, bugs_host());
+  ASSERT_TRUE(contains(execution_space_instances.at(1), count, 1));
+  ASSERT_TRUE(contains(execution_space_instances.at(1), bugs, 0));
+}
+
+// This test ensures that it's possible to build a Kokkos::Graph using
+// Kokkos::Experimental::create_graph without providing a closure, but giving an
+// execution space instance.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), create_graph_no_closure_with_exec) {
+  auto graph = Kokkos::Experimental::create_graph<TEST_EXECSPACE>(ex);
+
+  auto root = Kokkos::Impl::GraphAccess::create_root_ref(graph);
+
+  auto node = root.then_parallel_for(1, count_functor{count, bugs, 0, 0});
+
+  graph.submit(ex);
+
+  ASSERT_TRUE(contains(ex, count, 1));
+  ASSERT_TRUE(contains(ex, bugs, 0));
+}
+
+// This test ensures that it's possible to build a Kokkos::Graph using
+// Kokkos::Experimental::create_graph without any argument.
+// Note that part of this test has to be skipped if the test fixture is
+// not instantiated for the default execution space.
+TEST_F(TEST_CATEGORY_FIXTURE(graph), create_graph_no_arg) {
+  auto graph = Kokkos::Experimental::create_graph();
+
+  static_assert(std::is_same_v<typename decltype(graph)::execution_space,
+                               Kokkos::DefaultExecutionSpace>);
+
+  auto root = Kokkos::Impl::GraphAccess::create_root_ref(graph);
+
+  if constexpr (std::is_same_v<typename decltype(graph)::execution_space,
+                               TEST_EXECSPACE>) {
+    auto node = root.then_parallel_for(1, count_functor{count, bugs, 0, 0});
+  }
+
+  graph.submit(graph.get_execution_space());
+
+  if constexpr (std::is_same_v<typename decltype(graph)::execution_space,
+                               TEST_EXECSPACE>) {
+    ASSERT_TRUE(contains(graph.get_execution_space(), count, 1));
+    ASSERT_TRUE(contains(graph.get_execution_space(), bugs, 0));
+  }
```

Comment:
Same comment about readability of the condition.
I really dislike our nor asserting anything in an else-branch or skipping

Reply:
OK, so as you suggested this morning, this test should probably be skipped from the beginning if the `TEST_EXECSPACE` is not the same as `Kokkos::DefaultExecutionSpace` ?

Reply:
Unless there is a value in checking that is is legal to build a graph w/o an argument and that when it is legal to submit it as is.

Reply:
Technically they are different but I don;t think this is critical

Reply:
OK, I get your point. But it seems that it is indeed not critical at this point, so let's close.